src/app/public/theming/theme-if.directive.spec.ts

```diff
@@ -0,0 +1,85 @@
+import {
+  ComponentFixture,
+  TestBed
+} from '@angular/core/testing';
+
+import {
+  expect
+} from '@skyux-sdk/testing';
+
+import {
+  SkyThemeModule
+} from '../theme.module';
+
+import {
+  SkyThemeIfTestComponent
+} from './fixtures/theme-if-test.component';
+
+import {
+  SkyThemeSettings
+} from './theme-settings';
+
+import {
+  SkyTheme
+} from './theme';
+
+import {
+  SkyThemeMode
+} from './theme-mode';
+
+describe('ThemeIf directive', () => {
+  let fixture: ComponentFixture<SkyThemeIfTestComponent>;
+  const defaultThemeSettings = new SkyThemeSettings(SkyTheme.presets.default, SkyThemeMode.presets.light);
+  const modernThemeSettings = new SkyThemeSettings(SkyTheme.presets.modern, SkyThemeMode.presets.dark);
+  const getElements = (): NodeList => {
+    return fixture.debugElement.nativeElement.querySelectorAll('.sky-theme-if-test');
+  };
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        SkyThemeIfTestComponent
+      ],
+      imports: [
+        SkyThemeModule
+      ]
+    });
+    fixture = TestBed.createComponent(SkyThemeIfTestComponent);
+  });
+
+  it('should initialize the theme service with default settings', () => {
+    expect(getElements().length).toBe(0);
+  });
+
+  it('should work with the default theme', () => {
+    fixture.componentInstance.themeService.setTheme(defaultThemeSettings);
```

Comment:
There's nothing necessarily wrong with this, but you may be able to slim down your testing overhead by just mocking out the `SkyThemeService` entirely. You could remove the constructor provider from your fixture and do  something like this in your test setup:
```
  let mockThemeSvc: {
    settingsChange: BehaviorSubject<SkyThemeSettingsChange>
  };

  beforeEach(() => {
    mockThemeSvc = {
      settingsChange: new BehaviorSubject<SkyThemeSettingsChange>(
        {
          currentSettings: new SkyThemeSettings(
            SkyTheme.presets.default,
            SkyThemeMode.presets.light
          ),
          previousSettings: undefined
        }
      )
    };
    
    ...
    
    TestBed.configureTestingModule({
      ...
      providers: [
        {
          provide: SkyThemeService,
          useValue: mockThemeSvc
        }
      ]
      ...
    });
  });
 ```